spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/direct/DirectBigQueryRelation.java

```diff
@@ -63,6 +64,8 @@ public class DirectBigQueryRelation extends BigQueryRelation
   private final TableDefinition defaultTableDefinition;
   private final BigQueryRDDFactory bigQueryRDDFactory;
 
+  private Filter[] pushedFilters = new Filter[] {};
```

Comment:
It seems that the only usage for this field is in equals() and hashcode(), and in both cases we compare (or rely on) the result of `getCompiledFilter(pushedFilters)`. Please save the compiled filter instead - as we calculate it anyway, there's no reason to do it twice.

Reply:
Thanks for the review, addressed.